### PR TITLE
ADT GA SDK update

### PR DESCRIPTION
**Pull request by c-ryan-k**
- Regenerated SDK from 2020-10-31 swagger
- Added support for `--deadletter-sas-uri` in `dt route create` commands
- Removed `preview` tag from `az dt` command group
- Updated python minimum requirement to 3.6
- Added documentation to explain that `--properties` are required for `dt twin create` if the twin contains components
- Minor test updates

---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to the IoT extension!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] If introducing new functionality or modified behavior, are they backed by unit and integration tests?
- [ ] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [ ] Have **all** unit **and** integration tests passed locally? i.e. `pytest <project root> -vv`
- [ ] Have static checks passed using the .pylintrc and .flake8 rules? Look at the CI scripts for example usage.
- [ ] Have you made an entry in HISTORY.rst which concisely explains your feature or change?

### CODE REVIEW

File: azext_iot/tests/digitaltwins/test_dt_resource_lifecycle_int.py [hunk 1]
```diff
@@ -476,14 +482,14 @@ def assert_common_endpoint_attributes(
     )
     assert endpoint_output["resourceGroup"]
 
-    if tags:
-        assert endpoint_output["properties"]["tags"] == validate_key_value_pairs(tags)
-
     assert endpoint_output["properties"]["provisioningState"]
     assert endpoint_output["properties"]["createdTime"]
+    if dead_letter_secret:
+        import pdb; pdb.set_trace()
```
Comment (digimaun) [inline azext_iot/tests/digitaltwins/test_dt_resource_lifecycle_int.py:488]: breakpoint

File: azext_iot/tests/digitaltwins/test_dt_resource_lifecycle_int.py [hunk 2]
```diff
@@ -425,17 +423,25 @@ def test_dt_endpoints_routes(self):
...
-            ).get_output_in_json()
-            assert delete_ep_output["provisioningState"] == "Deleting"
-            assert delete_ep_output["id"].endswith("/{}".format(endpoint_name))
-            sleep(15)  # Wait for service to catch-up. Service will fix at some point.
+            )
+            # DT endpoint delete does not currently return data
+            # ).get_output_in_json()
+            # assert delete_ep_output["properties"]["provisioningState"] == "Deleting"
+            # assert delete_ep_output["id"].endswith("/{}".format(endpoint_name))
+            sleep(10)  # Wait for service to catch-up. Service will fix at some point.
```
Comment (c-ryan-k) [inline azext_iot/tests/digitaltwins/test_dt_resource_lifecycle_int.py:437]: FYI - Apparently there's a service update currently rolling out that should fix this, and endpoint delete (like dt delete) should in fact return an instance of the endpoint.

**Comment by c-ryan-k** [comment]
Integration tests (with new minimum component --properties test) and unit tests are green:
image

File: azext_iot/digitaltwins/params.py [hunk 3]
```diff
@@ -234,7 +235,8 @@ def load_digitaltwins_arguments(self, _):
             "properties",
             options_list=["--properties", "-p"],
             help="Initial property values for instantiating a digital twin or related components. "
-            "Provide file path or inline JSON.",
+            "Provide file path or inline JSON. Properties are required for twins that contain components, "
+            "at the minimum you must provide an object for each component that contains an empty $metadata object.",
```
Comment (digimaun) [inline azext_iot/digitaltwins/params.py:239]: nit, I think the second line could be slightly clearer like 

`at the minimum you must provide an empty $metadata object for each component.`

File: azext_iot/tests/digitaltwins/test_dt_resource_lifecycle_int.py (hunk 2 above)
Comment (c-ryan-k) [reply azext_iot/tests/digitaltwins/test_dt_resource_lifecycle_int.py:437]: Update - I misunderstood, endpoints (like DT instances) will finish their polling on DELETE with a 204: NoContent

**Comment by c-ryan-k** [comment]
After latest updates - tests still green
image

**Review by digimaun — APPROVED** [review]

**Pull request merged**